pkg/parser/custom/py/setuppy_parser.go

```diff
@@ -0,0 +1,233 @@
+package py 
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"context"
+	// "github.com/mpvl/unique"
+	"github.com/safedep/vet/pkg/common/logger"
+	treesitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/python"
+)
+
+// SetuppyParserViaSyntaxTree holds the parsing and extraction logic for setup.py files.
+type SetuppyParserViaSyntaxTree struct
+{
+	Symbol2strings map[string][]string
+	Symbol2symbols map[string][]string
+	Parser *treesitter.Parser
+}
+
+// NewSetuppyParserViaSyntaxTree creates a new instance of SetuppyParserViaSyntaxTree.
+func NewSetuppyParserViaSyntaxTree() *SetuppyParserViaSyntaxTree {
+	s := &SetuppyParserViaSyntaxTree{}
+	s.Symbol2strings = make(map[string][]string, 0)
+	s.Symbol2symbols = make(map[string][]string, 0)
+	s.Parser = treesitter.NewParser()
+	s.Parser.SetLanguage(python.GetLanguage())
+	return s
+}
+
+// extractStringConstants recursively extracts string constants and symbols from the syntax tree nodes.
+func (s *SetuppyParserViaSyntaxTree) extractStringConstants(node *treesitter.Node, code []byte) ([]string, []string) {
+	const_strings := []string{}
+	symbol_string := []string{}
+	// fmt.Println(" - ", string(node.Content(code)) )
+	// fmt.Println(node.ChildCount() , node.Type())
+	switch node.Type() {
+	case "identifier":
```

Comment:
Can we define these literals like `identifier`, `function_definition`, `string` as constants in this file assuming they are private to this parser?